File: Python/11.py
"""
 * IMPORTANTE: Sólo debes subir el fichero de código como parte del ejercicio.
 *
 * EJERCICIO:
 * Desarrolla un programa capaz de crear un archivo que se llame como
 * tu usuario de GitHub (Marioalf2002) y tenga la extensión .txt.
 * Añade varias líneas en ese fichero:
 * - Tu nombre.
 * - Edad.
 * - Lenguaje de programación favorito.
 * Imprime el contenido.
 * Borra el fichero.
 *
 * DIFICULTAD EXTRA (opcional):
 * Desarrolla un programa de gestión de ventas que almacena sus datos en un
 * archivo .txt.
 * - Cada producto se guarda en una línea del archivo de la siguiente manera:
 *   [nombre_producto], [cantidad_vendida], [precio].
 * - Siguiendo ese formato, y mediante terminal, debe permitir añadir, consultar,
 *   actualizar, eliminar productos y salir.
 * - También debe poseer opciones para calcular la venta total y por producto.
 * - La opción salir borra el .txt.
"""

# Creación de fichero
fichero = open("Marioalf2002.txt", "w")
fichero.write("Mario\n")
fichero.write("19\n")
fichero.write("Python\n")
fichero.close()

# Lectura de fichero
fichero = open("Marioalf2002.txt", "r")
print(fichero.read())
fichero.close()

# Borrado de fichero
import os
os.remove("Marioalf2002.txt")

# Dificultad extra
def anadir_producto(nombre, cantidad, precio):
    fichero = open("ventas.txt", "a")
    fichero.write(f"{nombre}, {cantidad}, {precio}\n")
    fichero.close()

def consultar_producto(nombre):
    fichero = open("ventas.txt", "r")
    for linea in fichero:
        if nombre in linea:
            print(linea)
    fichero.close()

def actualizar_producto(nombre, cantidad, precio):
    fichero = open("ventas.txt", "r")
    lineas = fichero.readlines()
    fichero.close()
    fichero = open("ventas.txt", "w")
    for linea in lineas:
        if nombre in linea:
            fichero.write(f"{nombre}, {cantidad}, {precio}\n")
        else:
            fichero.write(linea)
    fichero.close()

def eliminar_producto(nombre):
    fichero = open("ventas.txt", "r")
    lineas = fichero.readlines()
    fichero.close()
    fichero = open("ventas.txt", "w")
    for linea in lineas:
        if nombre not in linea:
            fichero.write(linea)
    fichero.close()

def venta_total():
    fichero = open("ventas.txt", "r")
    total = 0
    for linea in fichero:
        total += int(linea.split(", ")[1]) * float(linea.split(", ")[2])
    fichero.close()
    return total

def venta_producto(nombre):
    fichero = open("ventas.txt", "r")
    for linea in fichero:
        if nombre in linea:
            return int(linea.split(", ")[1]) * float(linea.split(", ")[2])
    fichero.close()

while True:
    print("1. Añadir producto")
    print("2. Consultar producto")
    print("3. Actualizar producto")
    print("4. Eliminar producto")
    print("5. Venta total")
    print("6. Venta por producto")
    print("7. Salir")
    opcion = int(input("Opción: "))

    if opcion == 1:
        nombre = input("Nombre: ")
        cantidad = int(input("Cantidad: "))
        precio = float(input("Precio: "))
        anadir_producto(nombre, cantidad, precio)
    elif opcion == 2:
        nombre = input("Nombre: ")
        consultar_producto(nombre)
    elif opcion == 3:
        nombre = input("Nombre: ")
        cantidad = int(input("Cantidad: "))
        precio = float(input("Precio: "))
        actualizar_producto(nombre, cantidad, precio)
    elif opcion == 4:
        nombre = input("Nombre: ")
        eliminar_producto(nombre)
    elif opcion == 5:
        print(venta_total())
    elif opcion == 6:
        nombre = input("Nombre: ")
        print(venta_producto(nombre))
    elif opcion == 7:
        os.remove("ventas.txt")
        break
    else:
        print("Opción no válida")

File: PhP/11.php
/*
* IMPORTANTE: Sólo debes subir el fichero de código como parte del ejercicio.
*
* EJERCICIO:
* Desarrolla un programa capaz de crear un archivo que se llame como
* tu usuario de GitHub (Marioalf2002) y tenga la extensión .txt.
* Añade varias líneas en ese fichero:
* - Tu nombre.
* - Edad.
* - Lenguaje de programación favorito.
* Imprime el contenido.
* Borra el fichero.
*
* DIFICULTAD EXTRA (opcional):
* Desarrolla un programa de gestión de ventas que almacena sus datos en un
* archivo .txt.
* - Cada producto se guarda en una línea del archivo de la siguiente manera:
* [nombre_producto], [cantidad_vendida], [precio].
* - Siguiendo ese formato, y mediante terminal, debe permitir añadir, consultar,
* actualizar, eliminar productos y salir.
* - También debe poseer opciones para calcular la venta total y por producto.
* - La opción salir borra el .txt.
*/

<?php

// Crear archivo
$archivo = fopen('Marioalf2002.txt', 'w');
fwrite($archivo, "Mario Alfredo\n");
fwrite($archivo, "19\n");
fwrite($archivo, "PhP\n");
fclose($archivo);

// Imprimir contenido
$archivo = fopen('Marioalf2002.txt', 'r');
while (!feof($archivo)) {
    echo fgets($archivo);
}
fclose($archivo);

// Borrar archivo
unlink('Marioalf2002.txt');

// DIFICULTAD EXTRA
function agregarProducto($nombre, $cantidad, $precio)
{
    $archivo = fopen('ventas.txt', 'a+');
    fwrite($archivo, "$nombre, $cantidad, $precio\n");
    fclose($archivo);
}

function consultarProductos()
{
    $archivo = fopen('ventas.txt', 'r');
    while (!feof($archivo)) {
        echo fgets($archivo);
    }
    fclose($archivo);
}

function actualizarProducto($nombre, $cantidad, $precio)
{
    $archivo = fopen('ventas.txt', 'r');
    $lineas = [];
    while (!feof($archivo)) {
        $lineas[] = fgets($archivo);
    }
    foreach ($lineas as &$linea) {
        $datos = explode(', ', $linea);
        if ($datos[0] === $nombre) {
            $linea = "$nombre, $cantidad, $precio\n";
            break;
        }
    }
    fclose($archivo);
    $archivo = fopen('ventas.txt', 'w');
    foreach ($lineas as $linea) {
        fwrite($archivo, $linea);
    }
    fclose($archivo);
}

function eliminarProducto($nombre)
{
    $archivo = fopen('ventas.txt', 'r');
    $lineas = [];
    while (!feof($archivo)) {
        $lineas[] = fgets($archivo);
    }
    foreach ($lineas as $indice => $linea) {
        $datos = explode(', ', $linea);
        if ($datos[0] === $nombre) {
            unset($lineas[$indice]);
            break;
        }
    }
    fclose($archivo);
    $archivo = fopen('ventas.txt', 'w');
    foreach ($lineas as $linea) {
        fwrite($archivo, $linea);
    }
    fclose($archivo);
}

function calcularVentaTotal()
{
    $archivo = fopen('ventas.txt', 'r');
    $ventaTotal = 0;
    while (!feof($archivo)) {
        $linea = fgets($archivo);
        $datos = explode(', ', $linea);
        $cantidad = intval($datos[1]);
        $precio = intval($datos[2]);
        $ventaTotal += $cantidad * $precio;
    }
    fclose($archivo);
    return $ventaTotal;
}

function calcularVentaPorProducto($nombre)
{
    $archivo = fopen('ventas.txt', 'r');
    $ventaProducto = 0;
    while (!feof($archivo)) {
        $linea = fgets($archivo);
        $datos = explode(', ', $linea);
        if ($datos[0] === $nombre) {
            $cantidad = intval($datos[1]);
            $precio = intval($datos[2]);
            $ventaProducto = $cantidad * $precio;
            break;
        }
    }
    fclose($archivo);
    return $ventaProducto;
}

// Ejemplo de uso
agregarProducto("Producto1", 10, 100);
agregarProducto("Producto2", 20, 200);
agregarProducto("Producto3", 30, 300);
consultarProductos();
actualizarProducto("Producto1", 20, 100);
eliminarProducto("Producto2");
consultarProductos();
$ventaTotal = calcularVentaTotal();
$ventaProducto = calcularVentaPorProducto("Producto3");
echo "Venta total: $ventaTotal\n";
echo "Venta del Producto3: $ventaProducto\n";

?>